x/emissions/keeper/msgserver/msg_server_worker_payload.go

```diff
@@ -107,11 +107,27 @@ func (ms msgServer) InsertWorkerPayload(ctx context.Context, msg *types.MsgInser
 				"Error forecaster address is not registered in this topic")
 		}
 
+		latestForecastScores := make(map[string]types.Score)
 		// Remove duplicate forecast element
 		acceptedForecastElements := make([]*types.ForecastElement, 0)
 		seenInferers := make(map[string]bool)
+
+		for _, el := range forecast.ForecastElements {
+			score, err := ms.k.GetLatestInfererScore(ctx, forecast.TopicId, el.Inferer)
+			if err != nil {
+				continue
+			}
+			latestForecastScores[el.Inferer] = score
```

Comment:
would instead call this `latestScoresForForecastedInferers` cuz otherwise it sounds like "forecaster scores" which are a different concept